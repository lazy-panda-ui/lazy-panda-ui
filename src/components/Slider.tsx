import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import SliderRN from '@react-native-community/slider';
import { useTheme } from '../theme/ThemeProvider';

export interface SliderProps {
  value: number;
  onValueChange: (value: number) => void;
  minimumValue?: number;
  maximumValue?: number;
  step?: number;
  style?: ViewStyle;
}

export const Slider: React.FC<SliderProps> = ({ value, onValueChange, minimumValue = 0, maximumValue = 100, step = 1, style }) => {
  const theme = useTheme();
  return (
    <View style={style}>
      <SliderRN
        value={value}
        onValueChange={onValueChange}
        minimumValue={minimumValue}
        maximumValue={maximumValue}
        step={step}
        minimumTrackTintColor={theme.colors.primary}
        maximumTrackTintColor={theme.colors.border}
        thumbTintColor={theme.colors.primary}
        style={styles(theme).slider}
      />
    </View>
  );
};

const styles = (theme: any) => StyleSheet.create({
  slider: {
    width: '100%',
    height: 40,
  },
});
