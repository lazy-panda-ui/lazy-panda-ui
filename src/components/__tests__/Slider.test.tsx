import React from 'react';
import { render } from '@testing-library/react-native';
import { Slider } from '../Slider';

describe('Slider', () => {
  it('renders with value', () => {
    const { toJSON } = render(
      <Slider value={0.5} onValueChange={function() {}} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
