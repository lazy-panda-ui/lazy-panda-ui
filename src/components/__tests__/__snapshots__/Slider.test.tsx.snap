// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders with value 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      [
        [
          {
            "height": 40,
          },
          {
            "height": 40,
            "width": "100%",
          },
        ],
        {
          "justifyContent": "center",
        },
      ]
    }
  >
    <RNCSlider
      disabled={false}
      inverted={false}
      lowerLimit={-9007199254740991}
      maximumTrackTintColor="#E5E5EA"
      maximumValue={100}
      minimumTrackTintColor="#007AFF"
      minimumValue={0}
      onChange={[Function]}
      onRNCSliderAccessibilityAction={null}
      onRNCSliderSlidingComplete={null}
      onRNCSliderSlidingStart={null}
      onRNCSliderValueChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      step={1}
      style={
        [
          {
            "width": 0,
            "zIndex": 1,
          },
          {
            "height": 40,
          },
          {
            "alignContent": "center",
            "alignItems": "center",
          },
        ]
      }
      tapToSeek={false}
      thumbImage={null}
      thumbTintColor="#007AFF"
      upperLimit={9007199254740991}
      value={0.5}
    />
  </View>
</View>
`;
